docs/configuration/configuration.md

```diff
@@ -2699,6 +2699,94 @@ tls_config:
+
+The private IP address is used for scraping by default, but may be changed to the public IP address with relabeling.
+
+The following meta labels are available on targets during [relabeling](#relabel_config):
+
+* `__meta_oci_compute_instance_id` : The unique ID for the OCI Compute Instance, its OCID.
+* `__meta_oci_compute_instance_name` : The display name for the instance.
+* `__meta_oci_compute_instance_state` : The lifecycle state of the instance.
+* `__meta_oci_compute_compartment` : The fully qualified compartment name for the instance.
+* `__meta_oci_compute_compartment_id` : The unique ID for the compartment for the instance.
+* `__meta_oci_compute_region` : OCI [region](https://docs.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm) code for this instance.
+* `__meta_oci_compute_availability_domain` : OCI [availability domain](https://docs.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm#top) code for this instance, within the region it resides in.
+* `__meta_oci_compute_fault_domain` : The OCI [fault domain](https://docs.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm#fault) code for this instance.
+* `__meta_oci_compute_vcn` : The VCN(virtual cloud network) display name for the instance, if available.
+* `__meta_oci_compute_vcn_id` : The OCID for the VCN for the instance, if available.
+* `__meta_oci_compute_subnet_id` : The OCID for the subnet for the instance.
+* `__meta_oci_compute_networking_bandwidth_in_gbps` : The alloted networking bandwidth in GBPS for the instance.
+* `__meta_oci_private_ip` : The private ip address of primary vnic for the instance.
+* `__meta_oci_public_ip` : The public ip address of primary vnic for the instance, if present.
+* `__meta_oci_compute_ocpus` : The OCPU count for the instance.
+* `__meta_oci_compute_processor_description` :The processor description for the instance.
+* `__meta_oci_compute_memory_in_gbs` : The amount of RAM in GBs.
+* `__meta_oci_compute_instance_shape` : The OCI compute [shape](https://docs.oracle.com/en-us/iaas/Content/Compute/References/computeshapes.htm) for the instance.
+* `__meta_oci_compute_defined_tag_<tagkey>` : Each [defined tag](https://docs.oracle.com/en-us/iaas/Content/Tagging/Concepts/taggingoverview.htm#concepts) value of the instance.
+* `__meta_oci_compute_freeform_tag_<tagkey>` : Each [freeform tag](https://docs.oracle.com/en-us/iaas/Content/Tagging/Concepts/understandingfreeformtags.htm) value of the instance.
+
+See below for the configuration options for *OCI Compute* discovery:
+
+```yaml
+# The OCID, the unique ID about your OCI tenancy/account. Always required.
+tenancy_ocid: <string> 
+# The OCI region code, 
+# as per https://docs.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm. This determines the OCI API endpoint used by OCI Compute SD. Always required.
+region: <string> 
+# The OCID for the compartment for your compute instances to be scraped.
+compartment_ocid: <string>
```

Comment:
You wonder this could either be a list or omitted maybe?

As to avoid duplicating the whole relabel config in case one has many compartments to scrape?

Reply:
Hi @neumayer 
We can not omit Compartment, since all OCI metadata API have Compartment-Id as must have field.
All resources in OCI Cloud are organized in Compartment. Think of Compartment as sub-account of AWS, org account in AWS under the same account(I do not exact AWS term).
Yes, we can make it a list, but there is no requirement from the customer as of now, since typically for a job it will be a single Compartment & conceptually another Compartment it will be mostly another job. & if in the future customers want it to be a list, we can update it at that time. Lmk what you think.

Reply:
If omitted you can query to get a list of all compartments. Or you can specify a list.

The example here is using the node exporter. It is not all that unlikely that users would have instances in multiple compartments. Those would share the same relabelling setup.

I'm not saying it's a must have, and we can work around this at the cost of some copy/paste.

I've been running your sd for a day now and it seems all good, and much easier to use than our previous solution (a script writing the json file for the file based sd). So it would clearly be worth it even with the copy/paste for many compartments (I don't think it's too exotic a use case).

Reply:
We can update/add support for list of compartments in the future if customers/users ask for it definitely.